modules/nf-core/software/fastqc/functions.nf

```diff
@@ -39,10 +39,17 @@ def getPathFromList(path_list) {
  */
 def saveFiles(Map args) {
     if (!args.filename.endsWith('.version.txt')) {
-        def ioptions = initOptions(args.options)
+        def ioptions  = initOptions(args.options)
         def path_list = [ ioptions.publish_dir ?: args.publish_dir ]
-        if (ioptions.publish_by_id) {
-            path_list.add(args.publish_id)
+        if (ioptions.publish_by_meta) {
+            def publish_id_list = ioptions.publish_by_meta instanceof List ? ioptions.publish_by_meta : args.publish_by_meta
+            for (id in publish_id_list) {
+                if (args.meta && id instanceof String) {
+                    def path = args.meta.containsKey(id) ? args.meta[id] : id
+                    path     = path instanceof String    ? path          : id
```

Comment:
Nope, It would end up as `WT_REP1/single_end` 💥 due to this line i.e. if the `path` isn't a string then use the value of `id` instead i.e. `single_end`

Reply:
I think I still don't get it: what would be the point then? wouldn't it be always `single_end` then, irrespective of the value of `meta.single_end`?

Reply:
It would indeed but because `single_end` is in the `map` the line below would evaluate to `false`

```
def path = args.meta.containsKey(id) ? args.meta[id] : id
```

which is why this line checks again and uses the `id` (or `single_end`) instead of `false`
```
path     = path instanceof String    ? path          : id
```

Reply:
ok, thanks for clarifying :+1: